src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+    if ((!this->niwStormActive) && (this->unkTimer260 == 0) && (this->niwType == ENNIW_TYPE_REGULAR)) {
+        if ((this->unknownState28E != 7) && (90000.0f != this->unk2BC.x)) {
+            this->unkTimer260 = 10;
+            this->sfxTimer1 = 30;
+            this->unk29E = 1;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+            this->unkTimer254 = 100;
+            this->unk2EC = 0;
+            EnNiw_SetupRunning(this);
+        }
+
+        if ((this->collider.base.acFlags & AC_HIT)) {
+            this->collider.base.acFlags &= ~AC_HIT;
+            // health gets used as a hit counter until cucco storm
+            if (this->actor.colChkInfo.health > 0) {
+                this->actor.colChkInfo.health--;
+            }
+
+            if ((D_80893460 == 0) && (this->actor.colChkInfo.health == 0)) {
+                // now you've done it
+                this->unkTimer254 = 100;
+                D_80893460 = 1;
+                this->unk298 = 0;
+                this->sfxTimer1 = 10000;
+                this->unk2A4.x = this->unk2B0.x = this->actor.world.pos.x;
+                this->unk2A4.y = this->unk2B0.y = this->actor.world.pos.y;
+                this->unk2A4.z = this->unk2B0.z = this->actor.world.pos.z;
+                this->unkTimer252 = this->unkTimer250 = this->unk298;
+
+                this->unk264[8] = 0.0f;
+                this->unk264[6] = 0.0f;
+                this->unk264[5] = 0.0f;
+                this->unk264[7] = 0.0f;
+                this->niwStormActive = true;
+                this->actionFunc = EnNiw_Trigger;
+                this->unk304 = 0.0f;
+                this->unk300 = 0.0f;
+                this->actor.speedXZ = 0.0f;
+
+            } else {
+                this->unkTimer260 = 10;
+                this->sfxTimer1 = 30;
+                this->unk29E = 1;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+                this->unkTimer254 = 100;
+                this->unk2EC = 0;
+                EnNiw_SetupRunning(this);
+            }
+        }
+    }
+}
+
+void EnNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnNiw* this = (EnNiw*)thisx;
+    s8 pad0;
+    s16 i;
+    ActorPlayer* player = PLAYER;
+    s16 pad1;
+    s16 featherCount;
+    Vec3f pos;
+    Vec3f spB8;
+    Vec3f spAC;
+    s32 pad2[9];
+    s16 temp29C;
+    f32 featherScale;
+    f32 camResult;
+    f32 floorHeight;
+    f32 dist = 20.0f;
+    s32 pad3;
+
+    this->unusedCounter28C++;
+
+    if (this->niwType == ENNIW_TYPE_UNK1) {
+        this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.parent->shape.rot.y;
+    }
+
+    if (this->unknownState28E != 0) {
+        this->unk264[9] = 0.0f;
+    }
+
+    if (this->unk29E != 0) {
+        featherCount = 20;
+        if (this->unk29E == 2) {
+            featherCount = 4;
+        }
+        for (i = 0; i < featherCount; ++i) {
```

Comment:
```suggestion
        for (i = 0; i < featherCount; i++) {
```
I'm not sure which is better. I've seen this version more often but feel free to do as you like